data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -305,3 +312,89 @@ def create_codebook(
+    actual_state_fips_codes: typing.List[str],
+    nation_expected: bool = True,
+    puerto_rico_expected: bool = True,
+    island_areas_expected: bool = True,
+    additional_fips_codes_not_expected: typing.List[str] = None,
+    dataset_name: str = None,
+) -> None:
+    """Check whether a list of state/territory FIPS codes match expectations.
+
+    Args:
+        actual_state_fips_codes (List of str): Actual state codes observed in data
+        nation_expected (bool, optional): Do you expect the nation (DC & states) to be
+            represented in data?
+        puerto_rico_expected (bool, optional): Do you expect PR to be represented in data?
+        island_areas_expected (bool, optional): Do you expect Island Areas to be represented in
+            data?
+        additional_fips_codes_not_expected (List of str, optional): Additional state codes
+            not expected in the data. For example, the data may be known to be missing
+            data from Maine and Wisconsin.
+        dataset_name (str, optional): The name of the data set, used only in printing an
+            error message. (This is helpful for debugging during parallel etl runs.)
+
+    Returns:
+        None: Does not return any values.
+
+    Raises:
+        ValueError: if lists do not match expectations.
+    """
+    # Setting default argument of [] here to avoid mutability problems.
+    if additional_fips_codes_not_expected is None:
+        additional_fips_codes_not_expected = []
+
+    # Cast input to a set.
+    actual_state_fips_codes_set = set(actual_state_fips_codes)
+
+    # Start with the list of all FIPS codes for all states and territories.
+    expected_states_set = set(get_state_fips_codes(settings.DATA_PATH))
+
+    # If nation (states and DC) are not expected to be included, remove it from the
+    # expected
+    # states set.
+    if not nation_expected:
+        expected_states_set = expected_states_set - set(TILES_NATION_FIPS_CODE)
+
+    # If Puerto Rico is not expected to be included, remove it from the expected
+    # states set.
+    if not puerto_rico_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_PUERTO_RICO_FIPS_CODE
+        )
+
+    # If island areas are not expected to be included, remove them from the expected
+    # states set.
+    if not island_areas_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_ISLAND_AREA_FIPS_CODES
+        )
+
+    # If additional FIPS codes are not expected to be included, remove them from the
+    # expected states set.
+    expected_states_set = expected_states_set - set(
+        additional_fips_codes_not_expected
+    )
+
+    dataset_name_phrase = (
+        f" for dataset `{dataset_name}`" if dataset_name is not None else ""
+    )
+
+    if expected_states_set != actual_state_fips_codes_set:
+        raise ValueError(
+            f"The states and territories in the data{dataset_name_phrase} are not "
```

Comment:
missing a space in line 389

Reply:
Oh, well, there will always be a weird-looking lack of a space on this. Because it'll either be `data{dataset_name_phrase} are` or `data {dataset_name_phrase}are` but if `dataset_name_phrase` is blank you don't want the text to read `The states and territories in the data  are not` with two spaces between data and are. 

Does that make sense? Or am I missing a cleaner way to do this.